samples/tfm/tfm_cert/README.rst

```diff
@@ -0,0 +1,222 @@
+************************************
+
+Use firmware update to update the application and TF-M firmware.
+For the image to be updatable, the firmware image version :kconfig:option:`CONFIG_MCUBOOT_IMAGE_VERSION` has to be updated to a higher version.
+See :ref:`ug_fw_update_image_versions_mcuboot_downgrade` for information on downgrade protection in MCUboot.
+
+To upload a new application image, build an application with an updated image version.
+
+.. code-block:: console
+
+    west build -b nrf5340dk_nrf5340_cpuapp_ns nrf/samples/tfm/tfm_cert -d build_update \
+    -DCONFIG_MCUBOOT_IMAGE_VERSION=\"1.2.3\"
+
+Then upload the new application image to the device.
+
+.. code-block:: console
+
+    mcumgr --conntype serial --connstring dev=/dev/ttyACM2,baud=115200,mtu=512 image list
+    mcumgr --conntype serial --connstring dev=/dev/ttyACM2,baud=115200,mtu=512 image upload build_update/zephyr/app_update.bin
+
+Once the new application image is uploaded, the hash of the image is shown in the image list.
+Flag the image to be tested on next reboot using its hash.
+
+.. code-block:: console
+
+    mcumgr --conntype serial --connstring dev=/dev/ttyACM2,baud=115200,mtu=512 image list
+    mcumgr --conntype serial --connstring dev=/dev/ttyACM2,baud=115200,mtu=512 image test <hash>
+
+Trigger the application update by initiating a reset.
+The verification of the image will happen during the update process.
+
+.. code-block:: console
+
+    mcumgr --conntype serial --connstring dev=/dev/ttyACM2,baud=115200,mtu=512 reset
+
+Bootloader firmware update
+**************************
+
+To upload a new bootloader image, build a bootloader targeting the correct bootloader slot with an updated firmware image version.
+The bootloader is placed in slot 0 by default, so enable building of the slot 1 bootloader.
+
+    west build -b nrf5340dk_nrf5340_cpuapp_ns nrf/samples/tfm/tfm_cert \
+    -DCONFIG_BUILD_S1_VARIANT=y \
+    -Dmcuboot_CONFIG_FW_INFO_FIRMWARE_VERSION=2
+
+List the current firmware images and upload a bootloader image that targets the non-active bootloader slot.
+
+.. code-block:: console
+
+    mcumgr --conntype serial --connstring dev=/dev/ttyACM2,baud=115200,mtu=512 image list
+    mcumgr --conntype serial --connstring dev=/dev/ttyACM2,baud=115200,mtu=512 image upload build/zephyr/signed_by_mcuboot_and_b0_s1_image_update.bin
+
+Once the new bootloader image is uploaded, the hash of the image is shown in the image list.
+Flag the image to be tested on next reboot using its hash.
+
+.. code-block:: console
+
+    mcumgr --conntype serial --connstring dev=/dev/ttyACM2,baud=115200,mtu=512 image list
+    mcumgr --conntype serial --connstring dev=/dev/ttyACM2,baud=115200,mtu=512 image test <hash>
+
+Trigger the bootloader update by initiating a reset.
+The verification of the image will happen during the update process.
+
+.. code-block:: console
+
+    mcumgr --conntype serial --connstring dev=/dev/ttyACM2,baud=115200,mtu=512 reset
+
+Testing
+=======
```

Comment:
This section is usually a subsection of Building and running. Maybe it should be moved to before the Firmware update section.